src/wp-includes/feed-atom-comments.php

```diff
@@ -27,20 +27,30 @@
 		do_action( 'atom_comments_ns' );
 	?>
 >
-	<title type="text">
 	<?php
+	$feed_title = '';
 	if ( is_singular() ) {
 		/* translators: Comments feed title. %s: Post title. */
-		printf( ent2ncr( __( 'Comments on %s' ) ), get_the_title_rss() );
+		$feed_title = sprintf( ent2ncr( __( 'Comments on: %s' ) ), get_the_title_rss() );
 	} elseif ( is_search() ) {
 		/* translators: Comments feed title. 1: Site title, 2: Search query. */
-		printf( ent2ncr( __( 'Comments for %1$s searching on %2$s' ) ), get_bloginfo_rss( 'name' ), get_search_query() );
+		$feed_title = sprintf( ent2ncr( __( 'Comments for %1$s searching on %2$s' ) ), get_bloginfo_rss( 'name' ), get_search_query() );
 	} else {
 		/* translators: Comments feed title. %s: Site title. */
-		printf( ent2ncr( __( 'Comments for %s' ) ), get_wp_title_rss() );
+		$feed_title = sprintf( ent2ncr( __( 'Comments for %s' ) ), get_bloginfo_rss( 'name' ) . get_wp_title_rss() );
 	}
+
+	/**
+	 * Filters the page title for the comments feed.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $feed_title Comments feed page title.
+	 */
+	$feed_title = apply_filters( 'comments_feed_title', $feed_title, get_the_title_rss() );
 	?>
-	</title>
+
+	<title type="text"><?php echo esc_html( $feed_title ); ?></title>
```

Comment:
Does `esc_html()` negate the need for `ent2ncr()`?

There was a comment on the ticket about late escaping, ie the no `ent2ncr()` when setting `$feed_title` and applying it upon echoing. This seems sensible to me as it will send the data to the filter in the original format (ie with any named entities that might be expected).

Reply:
> Does `esc_html()` negate the need for `ent2ncr()`?

I believe `ent2ncr()` is still needed, as named entities can break feeds and [should be converted to numeric entities](https://webmasters.stackexchange.com/questions/135624/how-to-substitute-html-entities-in-rss). `esc_html()` may or may not be redundant though.

Reply:
> I believe `ent2ncr()` is still needed, as named entities can break feeds

Thanks Sergey, I think it will need both in that case to run after the filter:

```
wp> ent2ncr( 'Pens & Pencils' );
string(14) "Pens & Pencils"
wp> ent2ncr( esc_html( 'Pens & Pencils' ) );
string(18) "Pens &#38; Pencils"
```